cmd/postgresql-upgrade-database.rb

```diff
@@ -66,8 +66,10 @@ def postgresql_upgrade_database
       # Following instructions from:
       # https://www.postgresql.org/docs/10/static/pgupgrade.html
       ohai "Upgrading #{name} data from #{pg_version_data} to #{pg_version_installed}..."
-
-      if /#{name}\s+started/.match?(Utils.popen_read("brew", "services", "list"))
+      services_json_output = Utils.popen_read("brew", "services", "info", "--all", "--json")
+      services_json = JSON.parse(services_json_output, object_class: OpenStruct)
+      service_names = services_json.map(&:name)
```

Comment:
You're not checking if it's started.

Reply:
<img width="486" alt="image" src="https://user-images.githubusercontent.com/24807638/155530105-4fb7e196-ee39-4c87-8a37-b7ed1501c49e.png">

postgres is trying to start and failing because the data directory hasn't been upgraded. so the service is not started. Loaded is true so we could select for that but still fix it for this particular scenario?

Reply:
I see, the script ensures that the original state of the service (trying to start, or actually stopped) is restored. So I guess putting the check for Loaded will achieve that?

Reply:
Yup, need to ensure it's either running or loaded.

Reply:
I see now adding the check for running services also would have been better!